applicationset/generators/matrix_test.go

```diff
@@ -1005,6 +1006,273 @@ func TestMatrixGenerateListElementsYaml(t *testing.T) {
+					"path[0]":                     "examples",
+					"path[1]":                     "git-generator-files-discovery",
+					"path[2]":                     "base",
+					"name":                        "prod-01",
+					"nameNormalized":              "prod-01",
+					"server":                      "https://prod-01.example.com",
+					"metadata.labels.environment": "prod",
+					"metadata.labels.argocd.argoproj.io/secret-type": "cluster",
+				},
+			},
+			clientError: false,
+			expectedMock: [][]map[string]interface{}{
+				{
+					{
+						"path":                        "examples/git-generator-files-discovery/base.json",
+						"path.basename":               "base",
+						"path.basenameNormalized":     "base",
+						"path.filename":               "base.json",
+						"path.filenameNormalized":     "base.json",
+						"path[0]":                     "examples",
+						"path[1]":                     "git-generator-files-discovery",
+						"path[2]":                     "base",
+					},
+				},
+			},
+			expectedNumCalls: 1,
+		},
+		{
+			name: "happy flow - generate interpolated params",
+			baseGenerators: []argoprojiov1alpha1.ApplicationSetNestedGenerator{
+				{ Clusters: interpolatedClusterGenerator },
+				{ Git: interpolatedGitGenerator },
+			},
+			expected: []map[string]interface{}{
+				{
+					"path":                        "examples/git-generator-files-discovery/dev-01.json",
+					"path.basename":               "dev-01",
+					"path.basenameNormalized":     "dev-01",
+					"path.filename":               "dev-01.json",
+					"path.filenameNormalized":     "dev-01.json",
+					"path[0]":                     "examples",
+					"path[1]":                     "git-generator-files-discovery",
+					"path[2]":                     "dev-01",
+					"name":                        "dev-01",
+					"nameNormalized":              "dev-01",
+					"server":                      "https://dev-01.example.com",
+					"metadata.labels.environment": "dev",
+					"metadata.labels.argocd.argoproj.io/secret-type": "cluster",
+				},
+				{
+					"path":                        "examples/git-generator-files-discovery/prod-01.json",
+					"path.basename":               "prod-01",
+					"path.basenameNormalized":     "prod-01",
+					"path.filename":               "prod-01.json",
+					"path.filenameNormalized":     "prod-01.json",
+					"path[0]":                     "examples",
+					"path[1]":                     "git-generator-files-discovery",
+					"path[2]":                     "prod-01",
+					"name":                        "prod-01",
+					"nameNormalized":              "prod-01",
+					"server":                      "https://prod-01.example.com",
+					"metadata.labels.environment": "prod",
+					"metadata.labels.argocd.argoproj.io/secret-type": "cluster",
+				},
+			},
+			clientError: false,
+			expectedMock: [][]map[string]interface{}{
+				{}, // Needed, because the 1st call will fail without interpolation
```

Comment:
I think this is true if and only if the generator contains a go template. The non-existence of a parameter does not cause `GenerateParams` to return nil or an error for fasttemplate fields. 

This problem is causing a test failure after another PR was merged:

```
--- FAIL: TestMatrixGenerate (0.00s)
    --- FAIL: TestMatrixGenerate/happy_flow_-_generate_params (0.00s)
        matrix_test.go:190: 
            	Error Trace:	/home/runner/work/argo-cd/argo-cd/applicationset/generators/matrix_test.go:190
            	Error:      	Not equal: 
            	            	expected: []map[string]interface {}{map[string]interface {}{"cluster":"Cluster", "path":"app1", "path.basename":"app1", "path.basenameNormalized":"app1", "templated":"test-app1", "url":"Url"}, map[string]interface {}{"cluster":"Cluster", "path":"app2", "path.basename":"app2", "path.basenameNormalized":"app2", "templated":"test-app2", "url":"Url"}}
            	            	actual  : []map[string]interface {}{map[string]interface {}{"cluster":"Cluster", "path":"app1", "path.basename":"app1", "path.basenameNormalized":"app1", "templated":"test-{{path.basenameNormalized}}", "url":"Url"}, map[string]interface {}{"cluster":"Cluster", "path":"app2", "path.basename":"app2", "path.basenameNormalized":"app2", "templated":"test-{{path.basenameNormalized}}", "url":"Url"}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -6,3 +6,3 @@
            	            	   (string) (len=23) "path.basenameNormalized": (string) (len=4) "app1",
            	            	-  (string) (len=9) "templated": (string) (len=9) "test-app1",
            	            	+  (string) (len=9) "templated": (string) (len=32) "test-{{path.basenameNormalized}}",
            	            	   (string) (len=3) "url": (string) (len=3) "Url"
            	            	@@ -14,3 +14,3 @@
            	            	   (string) (len=23) "path.basenameNormalized": (string) (len=4) "app2",
            	            	-  (string) (len=9) "templated": (string) (len=9) "test-app2",
            	            	+  (string) (len=9) "templated": (string) (len=32) "test-{{path.basenameNormalized}}",
            	            	   (string) (len=3) "url": (string) (len=3) "Url"
            	Test:       	TestMatrixGenerate/happy_flow_-_generate_params
FAIL
``` 

`requiresInterpolation` ends up being `false`, even though there's a templated field in the list generator.